From be0ee852a1bf5b5e755096dadaf43d1a9d3f2158 Mon Sep 17 00:00:00 2001
From: Hendrik Brueckner <brueckner@redhat.com>
Date: Wed, 2 Jul 2014 14:26:32 -0400
Subject: [s390] sclp_vt220: Enable ASCII console per default

Message-id: <1404311192-28323-1-git-send-email-brueckner@redhat.com>
Patchwork-id: 85180
O-Subject: [RHEL7.1 PATCH] [s390] s390/sclp_vt220: Enable ASCII console per default
Bugzilla: 1112739
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

Description
-----------
commit f2485f5d1c62bb8cccd452616892cdaf2466a090
Author: Peter Oberparleiter <oberpar@linux.vnet.ibm.com>
Date:   Wed Jun 4 17:01:57 2014 +0200

    s390/sclp_vt220: Enable ASCII console per default

    When you want to use the HMC's ASCII console as console device for
    a z/VM guest you have to specify console=ttyS1 on the kernel command
    line. But it won't work until you specify conmode=sclp as well.

    This behavior is inconsistent with the use of the ASCII console as
    TTY device which works on z/VM without the need to specify a conmode.

    Fix this inconsistency by removing the check for conmode=sclp in the
    ASCII console registration function.

    Signed-off-by: Peter Oberparleiter <oberpar@linux.vnet.ibm.com>
    Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>

Bugzilla
--------
BZ 1112739
https://bugzilla.redhat.com/show_bug.cgi?id=1112739

Upstream status of the patch
----------------------------
The patch is upstream as of kernel version
http://git.kernel.org/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commitdiff;h=f2485f5d1c62bb8cccd452616892cdaf2466a090

Brew
----
https://brewweb.devel.redhat.com/taskinfo?taskID=7656860

Test status
-----------
The patch has been tested and fixes the problem.
The fix has been verified by the IBM test department.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/s390/char/sclp_vt220.c b/drivers/s390/char/sclp_vt220.c
index 4eed38c..9a12db8 100644
--- a/drivers/s390/char/sclp_vt220.c
+++ b/drivers/s390/char/sclp_vt220.c
@@ -830,8 +830,6 @@ sclp_vt220_con_init(void)
 {
  int rc;
 
- if (!CONSOLE_IS_SCLP)
-  return 0;
  rc = __sclp_vt220_init(sclp_console_pages);
  if (rc)
   return rc;
-- 
1.7.1